app/models/solidus_stripe/payment_method.rb

```diff
@@ -7,37 +7,146 @@ class PaymentMethod < ::Spree::PaymentMethod
 
     validates :available_to_admin, inclusion: { in: [false] }
 
-    def partial_name
-      "stripe"
-    end
+    concerning :Actions do
+      def actions
+        %w[capture void credit]
+      end
+
+      def can_capture?(payment)
+        payment.pending?
+      end
 
-    alias cart_partial_name partial_name
-    alias product_page_partial_name partial_name
+      def can_void?(payment)
+        payment.pending?
+      end
 
-    def payment_source_class
-      PaymentSource
+      def can_credit?(payment)
+        payment.completed? && payment.credit_allowed > 0
+      end
     end
 
-    def gateway_class
-      Gateway
+    concerning :Configuration do
+      def partial_name
+        "stripe"
+      end
+
+      alias cart_partial_name partial_name
+      alias product_page_partial_name partial_name
+      alias risky_partial_name partial_name
+
+      def source_required?
+        false
+      end
+
+      def payment_source_class
+        PaymentSource
+      end
+
+      def gateway_class
+        Gateway
+      end
     end
 
-    def create_profile(payment)
-      payment_intent = payment.source.payment_intent
+    concerning :Order do
+      def find_in_progress_payment_for(order)
+        payments = order.payments
+          .where.not(state: %w[completed invalid void]) # in_progress
+          .where(payment_method: self)
+          .order(:created_at)
+          .entries
+
+        if payments.size > 1
+          payments[1...-1].each(&:invalidate!)
```

Comment:
If you mean to invalidate all potential existing payments and keep only the first one 
```suggestion
          payments[1..-1].each(&:invalidate!)
```

```ruby
[7] pry(#<SolidusStripe::PaymentMethod>)> a = [1,2,3]
[
    [0] 1,
    [1] 2,
    [2] 3
]
[8] pry(#<SolidusStripe::PaymentMethod>)> a[1..-1]
[
    [0] 2,
    [1] 3
]
[9] pry(#<SolidusStripe::PaymentMethod>)> a[1...-1]
[
    [0] 2
]
```

Reply:
Good catch! I actually meant to pick them all except the last one, but for some reason I used `1` instead of `0` as the first number, fixing!